src/java.base/share/classes/sun/security/util/Cache.java

```diff
@@ -591,4 +669,110 @@ public void invalidate() {
         }
     }
 
-}
+    /**
+     * This CacheEntry<K,V> type allows multiple V entries to be stored in
+     * one key in the cacheMap.
+     *
+     * This implementation is need for TLS clients that receive multiple
+     * PSKs or NewSessionTickets for server resumption.
+     */
+    private static class QueueCacheEntry<K,V> extends SoftReference<V>
+        implements CacheEntry<K,V> {
+
+        // Limit the number of queue entries.
+        private static final int MAXQUEUESIZE = 10;
+
+        final boolean DEBUG = false;
+        private K key;
+        private long expirationTime;
+        final Queue<CacheEntry<K,V>> queue = new ConcurrentLinkedQueue<>();
+
+        QueueCacheEntry(K key, V value, long expirationTime,
+            ReferenceQueue<V> queue) {
+            super(value, queue);
+            this.key = key;
+            this.expirationTime = expirationTime;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
```

Comment:
This method should call SoftReference.get() somewhere to let the GC know that this cache entry is still used